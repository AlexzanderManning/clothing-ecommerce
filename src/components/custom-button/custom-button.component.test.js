import { shallow } from "enzyme";
import "jest-enzyme";

import React from "react";
import CustomButton from "./custom.button.component";

describe("Custom Button Component", () => {
  const mockButton = {
    children: "button",
    isGoogleSignIn: "null",
    inverted: true,
    otherProps: [],
  };

  const wrapper = shallow(<CustomButton {...mockButton} />);

  it("renders to dom without crashing.", () => {
    expect.assertions(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("it is inverted", () => {
    expect.assertions(1);
    expect(wrapper.props().className.includes('inverted')).toEqual(true);
  });

  it("it is NOT inverted", () => {
    const mockButton = {
      children: "button",
      isGoogleSignIn: "null",
      inverted: null,
      otherProps: [],
    };
    const wrapper = shallow(<CustomButton {...mockButton} />);

    expect.assertions(1);
    expect(wrapper.props().className.includes("inverted")).toEqual(false);
  });

  it("it is a google sign in button", () => {
    expect.assertions(1);
    expect(wrapper.props().className.includes("google-sign-in")).toEqual(true);
  });

  it("it is NOT a google sign in button", () => {
    const mockButton = {
      children: "button",
      isGoogleSignIn: null,
      inverted: null,
      otherProps: [],
    };
    const wrapper = shallow(<CustomButton {...mockButton} />);
    expect.assertions(1);
    expect(wrapper.props().className.includes("google-sign-in")).toEqual(false);
  });
});
