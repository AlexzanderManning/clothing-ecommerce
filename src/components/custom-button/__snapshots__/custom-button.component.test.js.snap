// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Button Component renders to dom without crashing. 1`] = `
<button
  className="inverted google-sign-in custom-button"
  otherProps={Array []}
>
  button
</button>
`;
